Add release announcement blog article for Kubernetes v1.26

fsmunoz opened the pull request:
Release article for the v1.26 release, content based on the previously shared hackmd draft and selecion of identified major themes.

There is still ~two~ one areas that are to be added (the theme introduction ~and the use-cases description~), but the Major Themes can be reviewed.

fsmunoz commented:
/hold
/cc @leonardpahlke @ramrodo @krol3

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+### Progress on the CSI migration
+
+Following the GA of the [core Container Storage Interface (CSI) Migration](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration) feature in the previous release, CSI migration is an on-going effort that we've been working on for a few releases now, and this release continues to add (and remove) features aligned with the migration's goals.
```
Maybe link to the issue rather than the KEP - this'd match other items.

fsmunoz replied:
True, although in this case it was on purpose, because the KEP contains a broader description of the purpose, and v1.25 release article also linked to it. Since here it's more about setting the context, I think it works.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+#### In-tree GlusterFS driver removal
+
+Already deprecated in the v1.25 release, the [in-tree GlusterFS driver](https://github.com/kubernetes/enhancements/issues/3446) was removed in this release.
```
We're linking to the removal, not the driver, so:
```suggestion
Already deprecated in the v1.25 release, the in-tree GlusterFS driver was [removed](https://github.com/kubernetes/enhancements/issues/3446) in this release.
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+#### CSI migration for Azure File and vSphere graduated to Stable
+
+Both the [vSphere](https://github.com/kubernetes/enhancements/issues/1491) and [Azure](https://github.com/kubernetes/enhancements/issues/1885) in-tree driver migration to CSI have graduated to Stable.
```
Do we want to link to the docs for these storage drivers?

fsmunoz replied:
Good idea, I have added a link to both repos, take a look.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+### CRI v1alpha2 removed
+
+With the adoption of the [Container Runtime Interface](https://kubernetes.io/docs/concepts/architecture/cri/) (CRI) and the [removal of dockershim] in v1.24 , the CRI is the supported and documented way through which Kubernetes interacts with different container runtimes. Each kubelet negotiates which version of CRI to use with the container runtime on that node.
```
Is `removal of dockershim` a broken hyperlink?

fsmunoz replied:
It was, fixed.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+With the adoption of the [Container Runtime Interface](https://kubernetes.io/docs/concepts/architecture/cri/) (CRI) and the [removal of dockershim] in v1.24 , the CRI is the supported and documented way through which Kubernetes interacts with different container runtimes. Each kubelet negotiates which version of CRI to use with the container runtime on that node.
+
+The Kubernetes project recommends using CRI version `v1`; in Kubernetes v1.25 the kubelet could also negotiate the use of CRI `v1alpha2` (which was deprecated along at the same time as adding support for the stable `v1` interface).
```
It's a bit more than a recommend. That's now the only version we support, right?

fsmunoz replied:
Reworded it, it was meant to describe the previous situation but was not clear (I think it is now)

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+Kubernetes v1.26 dropped support for CRI `v1alpha2`. That [removal](https://github.com/kubernetes/kubernetes/pull/110618) will result in the kubelet not registering the node if the container runtime doesn't support CRI `v1`. This means that containerd minor version 1.5 and older will not be supported in Kubernetes 1.26; if you use containerd, you will need to upgrade to containerd version 1.6.0 o ...
+
+### Signing Kubernetes release artifacts graduates to Stable [SIG Release]
```
```suggestion
### Signing Kubernetes release artifacts graduates to Stable
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+#### Metrics framework extension graduates to Alpha
+
+The **metrics framework extension** graduates to Alpha and documentation can now be [auto-generated](https://kubernetes.io/docs/reference/instrumentation/metrics/) for every metric in the Kubernetes codebase, improving the ability to identify the metrics exposed by different Kubernetes components.
```
```suggestion
The **metrics framework extension** graduates to Alpha and [documentation](/docs/reference/instrumentation/metrics/) is now published for every metric in the Kubernetes codebase.
```
The autogeneration is largely a detail that readers don't need to know about.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+#### Component Service Level Indicators graduates to Alpha
+
+Also improving on the ability to consume Kubernetes metrics, the **component Service Level Indicators (SLIs)** has graduated to Alpha: by enabling the `ComponentSLIs` feature flag there will be an [additional metrics endpoint](https://github.com/kubernetes/kubernetes/pull/112884) which allows the calculation of Service Level Objectives (SLOs) from raw healthcheck data converted into metric format ...
```
```suggestion
Also improving on the ability to consume Kubernetes metrics, **component Service Level Indicators (SLIs)** have graduated to Alpha: by enabling the `ComponentSLIs` feature flag there will be an [additional metrics endpoint](https://github.com/kubernetes/kubernetes/pull/112884) which allows the calculation of Service Level Objectives (SLOs) from raw healthcheck data converted into metric format.
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+#### API server tracing
+
+_FIXME: API server tracing graduates to Beta [SIG Instrumentation] need clarification, https://github.com/kubernetes/enhancements/issues/647 indicates this is not tracked for 1.26_
```
This isn't done.

fsmunoz replied:
Indeed, will remove it - only kept it there because it was flagged as a major theme in the discussion and wanted to confirm.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+
+### Deprecations and Removals
+
+XX features were [deprecated or removed](/blog/2022/11/18/upcoming-changes-in-kubernetes-1-26/) from Kubernetes with this release. <FIXME: how many do we want to indicate here? Just the KEP? These two? More?>
```
This is not done yet.

fsmunoz replied:
Updated.

sftim reviewed (commented):
Thanks

ramrodo commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,190 @@
+#### `PodSchedulingReadiness` graduates to Alpha
+
+This feature introduces a `.spec.schedulingGates` field to Pod's API, to [indicate whether the Pod is allowed to be scheduled or not](https://github.com/kubernetes/enhancements/issues/3521
+). External users/controllers can use this field to hold a Pod from scheduling based on their policies and needs.
```
Not sure if this closing parenthesis in a new line will affect to the markdown syntax.
Should we move it to the end of the previous line?

```suggestion
External users/controllers can use this field to hold a Pod from scheduling based on their policies and needs.
```

jihoon-seo replied:
The link is displayed correctly on the rendered version (see [preview](https://deploy-preview-38123--kubernetes-io-main-staging.netlify.app/blog/2022/12/06/kubernetes-v1-26-release/#podschedulingreadiness-graduates-to-alpha)),
but I suggest to do so because
- some of the tools (e.g., [`linkchecker.py`](https://github.com/kubernetes/website/tree/main/scripts#linkcheckerpy)) might not recognize the markdown hyperlink codes with line-breaks.
- it will not be recognized as a complete markdown hyperlink code, preventing some smart features (e.g., IntelliSense) from working.
- ...

fsmunoz replied:
I will change it (the proposed commit looks like it removes the end paren instead of the newline).

ramrodo reviewed (commented):
Left a comment related to markdown syntax

fsmunoz commented:
I have added HostProcess to the themes since I think it's significant, would appreciate a review from SIG Windows on the short paragraph (which will then be expanded in the feature blog)

@puerco I also added a small paragraph on the registry change, referencing the recent blog. 
 
/cc @marosset
/cc @puerco

fsmunoz commented:
Added the theme intro.

Apart from nits and the outcome of further comments and reviews, the only thing holding this is the image (which will be added later this week and is a simple change).

/cc @leonardpahlke

marosset commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### Support for Windows privileged containers and graduates to Stable
+
+Privileged container support allows containers to run with similar access to the host as processes that run on the host directly. Support for this feature in Windows nodes, called HostProcess, will now [graduate to Stable](https://github.com/kubernetes/enhancements/issues/1981), enabling access to host resources (including network resources) from privileged containers.
```
```suggestion
Privileged container support allows containers to run with similar access to the host as processes that run on the host directly. Support for this feature in Windows nodes, called HostProcess Containers, will now [graduate to Stable](https://github.com/kubernetes/enhancements/issues/1981), enabling access to host resources (including network resources) from privileged containers.
```

marosset commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+Introduced in [v1.24](https://kubernetes.io/blog/2022/05/03/kubernetes-1-24-release-announcement/), this feature constitutes a significant milestone in improving the security of the Kubernetes release process. All release artifacts are signed using [cosign](https://github.com/sigstore/cosign/), and both binary artifacts and images [can be verified](https://kubernetes.io/docs/tasks/administer-clus ...
+
+### Support for Windows privileged containers and graduates to Stable
```
```suggestion
### Support for Windows privileged containers graduates to Stable
```

marosset commented:
> I have added HostProcess to the themes since I think it's significant, would appreciate a review from SIG Windows on the short paragraph (which will then be expanded in the feature blog)
> 
> @puerco I also added a small paragraph on the registry change, referencing the recent blog.
> 
> /cc @marosset /cc @puerco

Added two small suggested updates!

fsmunoz commented:
Thank you @marosset , committed

marosset commented:
Windows changes LGTM - Thanks @fsmunoz

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+It's with immense joy that we announce the release of Kubernetes v1.26!
+
+This release includes a total of 37 enhancements: eleven of them are graduating to Stable, ten are graduating to Beta, and sixteen of them are entering Alpha. We also have two features being deprecated or removed.
```
38 enhancements AFAIK.

2 features being deprecated.. AFAIK just one? which one have I missed? https://github.com/kubernetes/enhancements/issues/281

fsmunoz replied:
Yes, good catch, there was one that was added to the milestone a few days ago, will add it.

There's only one tracked enhancement in the board, but at the time of writing we were also aware of the GlusterFS removal, but the actual total is further down and it's 12, with API changes. We can either just mention the 1 tracked removal, or the total identified - either way I need to update this.

fsmunoz replied:
They aren't just smaller API changes though: we have the klog removal, GlusterFS removal, in-tree credential code, etc - they are all listed in the "Deprecations and Removals" section further down this article. There are 3 API removals (one of them with significant impact), 3 deprecations, the rest of the removals are not substantially different from one another in terms of relevance, which makes it difficult to separate. 

In terms of tracking, only the Dynamic Kubelet Configuration exists as a deprecation. In reality, we have (at least) the 12 identified ones.

One way around this is to mention the  3 "major" ones, which are exactly the ones that are described as a Major theme (CRI, GlusterFS, Cinder), and say something like:

> [...] We also have 12 deprecations and removals, 3 of which detailed below due to their impact.

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+The theme for Kubernetes v1.26 is _Electrifying_.
+
+Each Kubernetes release is the result of the coordinated effort of dedicated volunteers, and only made possible due to the use of a diverse and complex set of computing resources, spread out through multiple datacenters and regions worldwide. The end result of a release - the binaries, the image containers, the documentation - are then deployed on a growing number of personal, on-premises, and cl ...
```
...and only made possible due to the use of a diverse and complex set of computing resources...

IMO reads a bit like contributors are machines, lol

(feel free to rephrase my suggestion :D) 
```suggestion
Each Kubernetes release is the result of the coordinated effort of dedicated volunteers, supported by complex automation processes and tools, powered by a vast set of computing resources spread across multiple data centers and regions worldwide. The end result of this open source process are after each release - the Kubernetes binaries, container images, and documentation - which are then being used on a growing number of personal, on-premises, and cloud computing resources.
```

fsmunoz replied:
Ok, got your point, but OTOH I don't thing the suggested alternative really "flows"... what about this (I've removed some words which are not really needed to make

> Each Kubernetes release is the result of the coordinated effort of dedicated volunteers, supported by complex automation processes and tools that are powered by a vast set of computing resources spread across multiple data centers worldwide. The result of this open source process - the Kubernetes binaries, container images, documentation, and others - is then used on a growing number of personal, on-premises, and cloud computing resources.

leonardpahlke replied:
👍

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+## Major Themes
+    
+Kubernetes v1.26 is composed of many changes, brought to you by a worldwide team of volunteers. For this release, we have identified several major themes.
```
point out this is just a subset and reference a place where all the changes can be discovered? Maybe reference release notes?

leonardpahlke replied:
release notes are mentioned in a chapter below.. so scratch that :)

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### Change in container image registry
+
+In the previous release, [Kubernetes changed the container registry](/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/), allowing the spread of the load across multiple Cloud Providers and Regions, a change that reduced the reliance on a single entity and provided a faster download experience for a large number of users.
```
(and now under community infrastructure)

fsmunoz replied:
I've used the formulation of the previous blog. @puerco could you add your thoughts here?

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+In the previous release, [Kubernetes changed the container registry](/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/), allowing the spread of the load across multiple Cloud Providers and Regions, a change that reduced the reliance on a single entity and provided a faster download experience for a large number of users.
+
+This release will be the first one that will be exclusively published in the new [registry.k8s.io](http://registry.k8s.io) image registry: no tags from v1.26 will be published, and only tags from releases before v1.25 will continue to be updated, in the (now legacy) k8s.gcr.io image registry. Check [this recent article](/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) for more information on  ...
```
```suggestion
This release will be the first one that will be exclusively published in the new [registry.k8s.io](http://registry.k8s.io) image registry: no tags from v1.26 will be published, and only tags from releases before v1.25 will continue to be updated, in the legacy k8s.gcr.io container image registry.
```

remove the last sentence since we already mentioned the article the sentence before?

fsmunoz replied:
Yes, we should avoid the repetition. OTOH, I think it's important to point people towards the explanation. What I will do here is to replace the first link with the initial registry PR that gost merge in v1.25, and then keep the last sentence as an additional source of information. THis, I think works well - it's the first link that is out of context IMO.

jihoon-seo commented:
It would be great to change the filename from `2022-12-06-kubernetes-1.25-blog.md` to `2022-12-06-kubernetes-1.26-blog.md`.

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+Both the [vSphere](https://github.com/kubernetes/enhancements/issues/1491) and [Azure](https://github.com/kubernetes/enhancements/issues/1885) in-tree driver migration to CSI have graduated to Stable. You can find more information about them in the [vSphere CSI driver](https://github.com/kubernetes-sigs/vsphere-csi-driver) and [Azure File CSI driver](https://github.com/kubernetes-sigs/azurefile-c ...
+
+#### In-tree GlusterFS driver removal
+
+Already deprecated in the v1.25 release, the in-tree GlusterFS driver was [removed](https://github.com/kubernetes/enhancements/issues/3446) in this release.
+
+#### In-tree OpenStack Cinder driver removal
+
+This release removed the deprecated in-tree storage integration for OpenStack (the `cinder` volume type). You should migrate to external cloud provider and CSI driver from https://github.com/kubernetes/cloud-provider-openstack instead. For more information, visit [Cinder in-tree to CSI driver migration](https://github.com/kubernetes/enhancements/issues/1489).
```
I wonder if we shouldn't delete that, since we already announced that in our last blog post.

fsmunoz replied:
Both removals, or just one? We could remove both in the interest of being more concise (they are all listed in the Removals section). I've kept them since they are part of the CSI migration theme and since I grouped them, the impact is smaller. If we remove them we should reflect that in the previous proposal of how many removals we want to highlight at the very start - in that case, we would just have the CRI API one.

sftim replied:
We should leave these in. Don't assume readers read the previous blog article that announced upcoming removals.

leonardpahlke replied:
+1 removing both

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### Signing Kubernetes release artifacts graduates to Stable
+
+Introduced in [v1.24](https://kubernetes.io/blog/2022/05/03/kubernetes-1-24-release-announcement/), this feature constitutes a significant milestone in improving the security of the Kubernetes release process. All release artifacts are signed using [cosign](https://github.com/sigstore/cosign/), and both binary artifacts and images [can be verified](https://kubernetes.io/docs/tasks/administer-clus ...
```
KEP issue https://github.com/kubernetes/enhancements/issues/3031

fsmunoz replied:
Added.

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+#### Metrics framework extension graduates to Alpha
+
+The **metrics framework extension** graduates to Alpha and [documentation](/docs/reference/instrumentation/metrics/) is now published for every metric in the Kubernetes codebase.
```
1. ref kep issue : https://github.com/kubernetes/enhancements/issues/3498
2. Add one sentence mentioning `Internal` & `Beta` new stability levels: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3498-extending-stability#proposal

leonardpahlke replied:
@logicalhan perhaps you could have another look at this

fsmunoz replied:
Added that information in a new sentence. Maybe we should drop the documentation part? It seems relevant to me in the way it improves the ability to explore things, but maybe it's inconsequential to end users?

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+The **metrics framework extension** graduates to Alpha and [documentation](/docs/reference/instrumentation/metrics/) is now published for every metric in the Kubernetes codebase.
+
+#### Component Service Level Indicators graduates to Alpha
```
kep issue: https://github.com/kubernetes/enhancements/issues/3466 

maybe mentioning "Health SLIs" not just SLIs is more accurate

fsmunoz replied:
Done.

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### Dynamic Resource Allocation graduates to Alpha
+
+[Dynamic Resource Allocation](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3063-dynamic-resource-allocation/README.md) is a new feature that puts resource scheduling in the hands of 3rd-party developers: it moves away from the limited "countable" interface for requesting access to resources (e.g. `nvidia.com/gpu: 2`), providing an API more akin to that of persistent volume ...
```
PTAL @pohly

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+## Upcoming Release Webinar
+
+Join members of the Kubernetes v1.26 release team on <FIXME date> to learn about the major features of this release, as well as deprecations and removals to help plan for upgrades. 
```
@fsmunoz we can update the date now ;)

fsmunoz replied:
Yes, will do it at the end of today :D

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+In the previous release, [Kubernetes changed the container registry](https://github.com/kubernetes/kubernetes/pull/109938), allowing the spread of the load across multiple Cloud Providers and Regions, a change that reduced the reliance on a single entity and provided a faster download experience for a large number of users.
+
+This release will be the first one that will be exclusively published in the new [registry.k8s.io](http://registry.k8s.io) image registry: no tags from v1.26 will be published, and only tags from releases before v1.25 will continue to be updated, in the (now legacy) k8s.gcr.io image registry. Refer to [this recently published article](/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) for more  ...
```
One of these is right:
```suggestion
This release of Kubernetes is the first that is exclusively published in the new `registry.k8s.io` container image registry. In the (now legacy) `k8s.gcr.io` image registry, no container images tags for v1.26 will be published, and only tags from releases before v1.25 will continue to be updated. Refer to [registry.k8s.io: faster, cheaper and Generally Available](/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) for more information on the motivation, advantages, and implications of this significant change.
```
or
```suggestion
This release of Kubernetes is the first that is exclusively published in the new `registry.k8s.io` container image registry. In the (now legacy) `k8s.gcr.io` image registry, no container images tags for v1.26 will be published, and only tags from releases before v1.26 will continue to be updated. Refer to [registry.k8s.io: faster, cheaper and Generally Available](/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) for more information on the motivation, advantages, and implications of this significant change.
```

The difference is the text immediately following “only tags from releases before“

fsmunoz replied:
It's the first one, v1.25 will not be updated as well. Good change to the article reference, it looks much better with the title.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### CRI v1alpha2 removed
+
+With the adoption of the [Container Runtime Interface](https://kubernetes.io/docs/concepts/architecture/cri/) (CRI) and the [removal of dockershim](https://kubernetes.io/blog/2022/02/17/dockershim-faq/) in v1.24, the CRI is the only supported and documented way through which Kubernetes interacts with different container runtimes. Each kubelet negotiates which version of CRI to use with the contai ...
```
```suggestion
With the adoption of the [Container Runtime Interface](/docs/concepts/architecture/cri/) (CRI)
and the [removal of dockershim](/blog/2022/02/17/dockershim-faq/) in v1.24, the CRI is the
only supported and documented way through which Kubernetes interacts with different
container runtimes. Each kubelet negotiates which version of CRI to use with the container
runtime on that node.
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+In the previous release, the Kubernetes project recommended using CRI version `v1`, but kubelet could still negotiate the use of CRI `v1alpha2`, which was deprecated.
+
+Kubernetes v1.26 drops support for CRI `v1alpha2`. That [removal](https://github.com/kubernetes/kubernetes/pull/110618) will result in the kubelet not registering the node if the container runtime doesn't support CRI `v1`. This means that containerd minor version 1.5 and older will not be supported in Kubernetes 1.26; if you use containerd, you will need to upgrade to containerd version 1.6.0 or  ...
```
```suggestion
Kubernetes v1.26 drops support for CRI `v1alpha2`.
That [removal](https://github.com/kubernetes/kubernetes/pull/110618) will result in the
kubelet not registering the node if the container runtime doesn't support CRI `v1`. This
means that containerd minor version 1.5 and older are not supported in Kubernetes
1.26; if you use containerd, you will need to upgrade to containerd version 1.6.0 or
later **before** you upgrade that node to Kubernetes v1.26. This applies equally to
any other container runtimes that only support the `v1alpha2`: if that affects you,
you should contact the container runtime vendor for advice or check their website
for additional instructions in how to move forward.
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+Following the GA of the [core Container Storage Interface (CSI) Migration](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration) feature in the previous release, CSI migration is an on-going effort that we've been working on for a few releases now, and this release continues to add (and remove) features aligned with the migration's goals.
+
+#### Delegate FSGroup to CSI Driver graduated to Stable
```
How about:
```suggestion
#### _Delegate FSGroup to CSI Driver_ graduated to stable
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+#### Delegate FSGroup to CSI Driver graduated to Stable
+
+This feature allows Kubernetes to [supply the pod's `FSGroup` to the CSI driver when a volume is mounted](https://github.com/kubernetes/enhancements/issues/2317) so that the driver can utilize mount options to control volume permissions. Previously, the kubelet would always apply the `fsGroup`ownership and permission change to files in the volume according to the policy specified  in the `Pod.spe ...
```
```suggestion
This feature allows Kubernetes to
[supply the pod's `fsGroup` to the CSI driver when a volume is mounted](https://github.com/kubernetes/enhancements/issues/2317)
so that the driver can utilize mount options to control volume permissions.
Previously, the kubelet would always apply the `fsGroup`ownership and permission change
to files in the volume according to the policy specified  in the Pod's
`.spec.securityContext.fsGroupChangePolicy` field.
Starting with this release, CSI drivers have the option to apply the `fsGroup` settings during
attach or mount time of the volumes.
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+This feature allows Kubernetes to [supply the pod's `FSGroup` to the CSI driver when a volume is mounted](https://github.com/kubernetes/enhancements/issues/2317) so that the driver can utilize mount options to control volume permissions. Previously, the kubelet would always apply the `fsGroup`ownership and permission change to files in the volume according to the policy specified  in the `Pod.spe ...
+
+#### CSI migration for Azure File and vSphere graduated to Stable
```
```suggestion
#### CSI migration for Azure File and vSphere graduated to stable
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+This release removed the deprecated in-tree storage integration for OpenStack (the `cinder` volume type). You should migrate to external cloud provider and CSI driver from https://github.com/kubernetes/cloud-provider-openstack instead. For more information, visit [Cinder in-tree to CSI driver migration](https://github.com/kubernetes/enhancements/issues/1489).
+
+### Signing Kubernetes release artifacts graduates to Stable
```
```suggestion
### Signing Kubernetes release artifacts graduates to beta
```
(https://github.com/kubernetes/enhancements/issues/3031 says it's moving to beta?

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### Signing Kubernetes release artifacts graduates to Stable
+
+Introduced in Kubernetes v1.24, [this feature](https://github.com/kubernetes/enhancements/issues/3031) constitutes a significant milestone in improving the security of the Kubernetes release process. All release artifacts are signed using [cosign](https://github.com/sigstore/cosign/), and both binary artifacts and images [can be verified](https://kubernetes.io/docs/tasks/administer-cluster/verify ...
```
```suggestion
Introduced in Kubernetes v1.24, [artifact signing](https://github.com/kubernetes/enhancements/issues/3031) constitutes a significant milestone in improving the security of the Kubernetes release process. All release artifacts are signed using [cosign](https://github.com/sigstore/cosign/), and both binary artifacts and images can be [verified](/docs/tasks/administer-cluster/verify-signed-images/).
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+Introduced in Kubernetes v1.24, [this feature](https://github.com/kubernetes/enhancements/issues/3031) constitutes a significant milestone in improving the security of the Kubernetes release process. All release artifacts are signed using [cosign](https://github.com/sigstore/cosign/), and both binary artifacts and images [can be verified](https://kubernetes.io/docs/tasks/administer-cluster/verify ...
+
+### Support for Windows privileged containers graduates to Stable
```
```suggestion
### Support for Windows privileged containers graduates to stable
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### Support for Windows privileged containers graduates to Stable
+
+Privileged container support allows containers to run with similar access to the host as processes that run on the host directly. Support for this feature in Windows nodes, called HostProcess containers, will now [graduate to Stable](https://github.com/kubernetes/enhancements/issues/1981), enabling access to host resources (including network resources) from privileged containers.
```
```suggestion
Privileged container support allows containers to run with similar access to the host as
processes that run on the host directly. The Windows node implementation of this,
_HostProcess containers_, is generally available in Kubernetes 1.26. A HostProcess 
container lets a Pod access host hardware and other components (such as the
host network stack).
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+#### Component Health Service Level Indicators graduates to Alpha
+
+Also improving on the ability to consume Kubernetes metrics, [component health Service Level Indicators (SLIs)](https://github.com/kubernetes/enhancements/issues/3466) have [graduated to Alpha](https://github.com/kubernetes/kubernetes/pull/112884): by enabling the `ComponentSLIs` feature flag there will be an additional metrics endpoint which allows the calculation of Service Level Objectives (SL ...
```
(opinion) Are these really a major theme? They're both only alpha.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### Dynamic Resource Allocation graduates to Alpha
+
+[Dynamic Resource Allocation](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3063-dynamic-resource-allocation/README.md) is a new feature that puts resource scheduling in the hands of 3rd-party developers: it moves away from the limited "countable" interface for requesting access to resources (e.g. `nvidia.com/gpu: 2`), providing an API more akin to that of persistent volume ...
```
```suggestion
[Dynamic Resource Allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/)
is a new alpha feature that puts resource scheduling in the hands of 3rd-party developers:
it moves away from the limited "countable" interface for requesting access to resources
(e.g. `nvidia.com/gpu: 2`), providing an API more akin to that of [persistent volumes](/docs/concepts/storage/persistent-volumes/).
Under the hood, it uses the [Container Device Interface](https://github.com/container-orchestrated-devices/container-device-interface) (CDI)
to do its device injection.
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### CEL in Admission Control graduates to Alpha
+
+This feature introduces a `v1alpha1` API for [validating admission policies](https://github.com/kubernetes/enhancements/issues/3488), enabling extensible admission control via [Common Expression Language](https://github.com/google/cel-spec) expressions. Currently, custom policies are enforced via [admission webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-con ...
```
Features don't exactly **graduate** to alpha; instead, they arrive as a _new_ alpha feature.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+#### `PodSchedulingReadiness` graduates to Alpha
+
+This feature introduces a `.spec.schedulingGates` field to Pod's API, to [indicate whether the Pod is allowed to be scheduled or not](https://github.com/kubernetes/enhancements/issues/3521). External users/controllers can use this field to hold a Pod from scheduling based on their policies and needs.
```
Features don't exactly **graduate** to alpha; instead, they arrive as a _new_ alpha feature.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+Kubernetes v1.26 introduces some relevant enhancements to the ability to better control scheduling behavior.
+
+#### `PodSchedulingReadiness` graduates to Alpha
```
```suggestion
#### Alpha support for Pod scheduling readiness
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+By specifying a `nodeInclusionPolicy` in `topologySpreadConstraints`, you can control whether to [take taints/tolerations into consideration](https://github.com/kubernetes/enhancements/issues/3094) when calculating Pod Topology Spread skew.
+
+#### `MatchLabelKeysInPodTopologySpread` graduates to Beta
```
```suggestion
#### Label matching for pod topology spread constraints graduates to beta
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+This feature introduces a `.spec.schedulingGates` field to Pod's API, to [indicate whether the Pod is allowed to be scheduled or not](https://github.com/kubernetes/enhancements/issues/3521). External users/controllers can use this field to hold a Pod from scheduling based on their policies and needs.
+
+#### `NodeInclusionPolicyInPodTopologySpread` graduates to Beta
```
```suggestion
#### `nodeInclusionPolicy` for pod topology spread constraints graduates to beta
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+## Other Updates
+
+### Graduations to Stable
```
```suggestion
### Graduations to stable
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+* [CPUManager](https://github.com/kubernetes/enhancements/issues/3570)
+* [DeviceManager](https://github.com/kubernetes/enhancements/issues/3573)
+
+### Deprecations and Removals
```
```suggestion
### Deprecations and removals
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+In this release we want to recognise the importance of all these building blocks on which Kubernetes is developed and used, while at the same time raising awareness on the importance of taking the energy consumption footprint into account: environmental sustainability is an inescapable concern of creators and users of any software solution, and the environmental footprint of sofware, like Kuberne ...
+
+As a community, we always work to make each new release process better than before (in this release, we have [started to use Projects for tracking enhancements](https://github.com/orgs/kubernetes/projects/98/views/1), for example). If v1.24 "Stargazer" [had us looking upwards, to what is possible when our community comes together](/blog/2022/05/03/kubernetes-1-24-release-announcement/), and v1.25 ...
```
I'd make “v1.24 Stargazer” and “v1.25 Combiner“ be the hyperlinks, and put the text after that into italics.

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+As a community, we always work to make each new release process better than before (in this release, we have [started to use Projects for tracking enhancements](https://github.com/orgs/kubernetes/projects/98/views/1), for example). If v1.24 "Stargazer" [had us looking upwards, to what is possible when our community comes together](/blog/2022/05/03/kubernetes-1-24-release-announcement/), and v1.25 ...
+
+## Major Themes
```
```suggestion
## Major themes
```

klueska commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+
+### Dynamic Resource Allocation graduates to Alpha
+
+[Dynamic Resource Allocation](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3063-dynamic-resource-allocation/README.md) is a new feature that puts resource scheduling in the hands of 3rd-party developers: it moves away from the limited "countable" interface for requesting access to resources (e.g. `nvidia.com/gpu: 2`), providing an API more akin to that of persistent volume ...
```
```suggestion
[Dynamic Resource Allocation](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3063-dynamic-resource-allocation/README.md) is a new feature that puts resource scheduling in the hands of third-party developers, It offers an alternative to the "countable" interface for requesting access to resources (e.g. `nvidia.com/gpu: 2`), providing an API more akin to that of persistent volumes. Under the hood, it uses the [Container Device Interface](https://github.com/container-orchestrated-devices/container-device-interface) (CDI) to do its device injection. This feature is blocked by the `DynamicResourceAllocation` feature gate.
```

fsmunoz replied:
I was changing the filename at the same time, but I reflected the above changes in the latest commit, thanks! Take a look and lgtm if you think it's fine.

klueska replied:
LGTM

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+* Join the community discussion on [Discuss](https://discuss.kubernetes.io/)
+* Join the community on [Slack](http://slack.k8s.io/)
+* Post questions (or answer questions) on [Server Fault](https://serverfault.com/questions/tagged/kubernetes).
+* Share your Kubernetes [story](https://docs.google.com/a/linuxfoundation.org/forms/d/e/1FAIpQLScuI7Ye3VQHQTwBASrgkjQDSS5TP0g3AXfFhwSM9YpHgxRKFA/viewform)
```
```suggestion
* [Share](https://docs.google.com/a/linuxfoundation.org/forms/d/e/1FAIpQLScuI7Ye3VQHQTwBASrgkjQDSS5TP0g3AXfFhwSM9YpHgxRKFA/viewform) your Kubernetes story
```

sftim commented on content/en/blog/_posts/2022-12-06-kubernetes-1.25-blog.md:
```diff
@@ -0,0 +1,205 @@
+* Follow us on Twitter [@Kubernetesio](https://twitter.com/kubernetesio) for the latest updates
+* Join the community discussion on [Discuss](https://discuss.kubernetes.io/)
+* Join the community on [Slack](http://slack.k8s.io/)
+* Post questions (or answer questions) on [Server Fault](https://serverfault.com/questions/tagged/kubernetes).
```
```suggestion
* Post questions (or answer questions) on [Server Fault](https://serverfault.com/questions/tagged/kubernetes)
```

fsmunoz replied:
Done, but is it a style guideline not to use final periods in lists?

sftim replied:
It's fine either way but be consistent within one page

sftim reviewed (commented):
Hi.

We'd prefer to write headings in sentence case.

I've added some inline feedback too.

leonardpahlke commented:
> It would be great to change the filename from `2022-12-06-kubernetes-1.25-blog.md` to `2022-12-06-kubernetes-1.26-blog.md`.

good catch, we should rename the file

fsmunoz commented:
@sftim @leonardpahlke I changed the filename, but the reviews were made on the previous file. I will try to reflect the suggestions manually, and then resolve the conversation - please take a look after that is done to check if it's adequate.

@jihoon-seo filename changes, ty.

fsmunoz commented on content/en/blog/_posts/2022-12-06-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,210 @@
+
+It's with immense joy that we announce the release of Kubernetes v1.26!
+
+This release includes a total of 37 enhancements: eleven of them are graduating to Stable, ten are graduating to Beta, and sixteen of them are entering Alpha. We also have two features being deprecated or removed.
```
@leonardpahlke adding comments here: we should decide on the approach here:

1. "12 features removed or deprectated", aligned with the list we have at the end of the article
2. "1 feature removed or deprecated", aligned with the tracked enhancements (but weird, because we then go on detailing 3 removals)
3. "12 features removed or deprecated, including 3 that we will detail below", mixing the total with the 3 that we have mentioned (CRI + 2 storage drivers)
3b. Same, but if removing something from the major themes, update the number.

leonardpahlke replied:
nr. 3

fsmunoz commented on content/en/blog/_posts/2022-12-06-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,210 @@
+
+The theme for Kubernetes v1.26 is _Electrifying_.
+
+Each Kubernetes release is the result of the coordinated effort of dedicated volunteers, and only made possible due to the use of a diverse and complex set of computing resources, spread out through multiple datacenters and regions worldwide. The end result of a release - the binaries, the image containers, the documentation - are then deployed on a growing number of personal, on-premises, and cl ...
```
@leonardpahlke I changed this to something along the lines of what you proposed, with some changes to make it simpler.

leonardpahlke replied:
👍

fsmunoz commented on content/en/blog/_posts/2022-12-06-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,210 @@
+
+#### Metrics framework extension graduates to alpha
+
+The metrics framework extension [graduates to Alpha](https://github.com/kubernetes/enhancements/issues/3498), and [documentation](/docs/reference/instrumentation/metrics/) is now published for every metric in the Kubernetes codebase. This enhancement adds two additional metadata fields to Kubernetes metrics: `Internal` and `Beta`, representing different stages of metric maturity.
```
@leonardpahlke added the proposed sentence.

fsmunoz commented on content/en/blog/_posts/2022-12-06-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,210 @@
+
+#### Component Health Service Level Indicators graduates to alpha
+
+Also improving on the ability to consume Kubernetes metrics, [component health Service Level Indicators (SLIs)](https://github.com/kubernetes/enhancements/issues/3466) have [graduated to Alpha](https://github.com/kubernetes/kubernetes/pull/112884): by enabling the `ComponentSLIs` feature flag there will be an additional metrics endpoint which allows the calculation of Service Level Objectives (SL ...
```
@leonardpahlke added "Health" and the KEP.

leonardpahlke commented on content/en/blog/_posts/2022-12-06-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,210 @@
+#### In-tree OpenStack Cinder driver removal
+
+This release removed the deprecated in-tree storage integration for OpenStack (the `cinder` volume type). You should migrate to external cloud provider and CSI driver from https://github.com/kubernetes/cloud-provider-openstack instead. For more information, visit [Cinder in-tree to CSI driver migration](https://github.com/kubernetes/enhancements/issues/1489).
+
```
we can leave it in if we identified this as major theme. I was just wondering if we wanted to make it more concise and since this is already mentioned in this blog separately (at the bottom of the list) & in the Deprecation blog I guess no information will be lost.

fsmunoz replied:
Will keep them for now, but leave this unresolved for a couple of days more so we can take a new look with other changes in place.

humblec commented on content/en/blog/_posts/2022-12-06-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,210 @@
+Previously, the kubelet would always apply the `fsGroup`ownership and permission change to files in the volume according to the policy specified  in the Pod's `.spec.securityContext.fsGroupChangePolicy` field.
+Starting with this release, CSI drivers have the option to apply the `fsGroup` settings during attach or mount time of the volumes.
+
+This feature allows Kubernetes to [supply the pod's `FSGroup` to the CSI driver when a volume is mounted](https://github.com/kubernetes/enhancements/issues/2317) so that the driver can utilize mount options to control volume permissions. Previously, the kubelet would always apply the `fsGroup`ownership and permission change to files in the volume according to the policy specified  in the `Pod.spe ...
```
@xing-yang looks like line 53 to 55 got repeated from 57.

fsmunoz replied:
Indeed, fixing it.

fsmunoz commented on content/en/blog/_posts/2022-12-06-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,210 @@
+
+Following the GA of the [core Container Storage Interface (CSI) Migration](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/625-csi-migration) feature in the previous release, CSI migration is an on-going effort that we've been working on for a few releases now, and this release continues to add (and remove) features aligned with the migration's goals.
+
+#### _Delegate FSGroup to CSI Driver_ graduated to stable
```
Thank you @xing-yang , that's an important point. Would it work if the section was named "CSI migration and improvements", with the necessary adjustments in the intro paragraph? Or is moving it outside of the section better?

sftim replied:
“Storage improvements”?

fsmunoz replied:
That's a good formulation,will use it.

sftim commented:
For the blog team, I / we are happy with the intent here and can approve once it's ready.

klueska commented on content/en/blog/_posts/2022-12-08-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,208 @@
+
+### Dynamic Resource Allocation graduates to alpha
+
+[Dynamic Resource Allocation](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3063-dynamic-resource-allocation/README.md) is a new feature that puts resource scheduling in the hands of third-party developers: it offers an alternative to the limited "countable" interface for requesting access to resources (e.g. `nvidia.com/gpu: 2`), providing an API more akin to that of persis ...
```
Does it make more sense to link to the upcoming documentation for this feature than the KEP?

fsmunoz replied:
We're using both the KEPs and the issues throughout the article (even because in some cases documentation can still be wip), do you think in this case it's really needed to change it? and is the documentation URL for this "stable"?

sftim replied:
I do think that people reading the article will expect to see links to documentation rather than to design documents or enhancements trackers.
SIG Docs adds redirects when we reorganize the website, so links should stay stable or at least continue to work even if we restructure things.

The blog post will go out _after_ the new docs are live. If not, we're doing it wrong.

sftim replied:
Related: it'd be nice to edit the issue description of graduated enhancements to include a link to the relevant docs. But we don't do this yet.

klueska replied:
I believe the documentation should be a stable link. This was the PR where it was merged: https://github.com/kubernetes/website/pull/37766

I'll let you guys be the judge as to which is more appropriate -- the KEP is also linked from the documentation, for reference.

 Independently, will the blog post for this feature also be linked here somewhere? From what I understand having a blog post ready was a pre-requisite for being included in the list for this blog post.

sftim replied:
You **don't** have to write a post-release blog article to have your feature mentioned in either:
- the release notes
- the release announcement.

You do have to have that article ready on time to be included in the _post release blog article series_.

fsmunoz replied:
I can take a look at adding docs (I've based my work on previous examples and it wasn't common, hence the omission). In this case, if I wanted to add the URL (and eventually to the Graduated ones), it would have to be added during the short period after the docs are updated (and the URL is live) and this one is released - my only concern here is pushing something that could open more discussion (change the link description, use a different link word, etc) so close to the publication date.

sftim replied:
You can add the links now. They're on the `dev-1.26` branch and there's a preview at https://deploy-preview-37810--kubernetes-io-main-staging.netlify.app/

If docs aren't done yet (on that branch), don't link to them. No worries there. For this release in particular it feels to me that we are struggling to get the docs ready in time.

sftim replied:
https://github.com/kubernetes/website/pull/38123#discussion_r1037973299 describes a nice to have and is unimportant.

fsmunoz replied:
Thank you, will have a look and make some consistent changes during the weekend - will try to be limited in them.

fsmunoz replied:
@sftim @klueska could you take a look? I opted for using the docs to the term, and the link to the KEP when the "new feature" is mentioned.

sftim replied:
> I opted for using the docs to the term, and the link to the KEP when the "new feature" is mentioned.

That sounds good to me.

fsmunoz replied:
I've added more links, using a similar approach, to other additions.

fsmunoz commented:
All sections done and no missing information, so removing WIP tag; keeping hold since there is still some discussion that can be done through additional reviews, plus confirmation of the Webinar link.

/retitle Add release announcement blog article for Kubernetes v1.26

sftim commented:
We should keep the hold until the v1.26 release has actually happened.

leonardpahlke commented:
/hold

sftim reviewed (commented):
Localization teams will thank you if you wrap the lines in Markdown at say 100 characters.

fsmunoz commented:
@sftim would had heading anchors also be a good practice (we did it in the mid-cycle one)?

sftim commented:
> @sftim would had heading anchors also be a good practice (we did it in the mid-cycle one)?

I explicitly set those only when the heading text is particularly long, or if I anticipate a future change to the heading text.

fsmunoz commented on content/en/blog/_posts/2022-12-08-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,208 @@
+
+In this release we want to recognise the importance of all these building blocks on which Kubernetes is developed and used, while at the same time raising awareness on the importance of taking the energy consumption footprint into account: environmental sustainability is an inescapable concern of creators and users of any software solution, and the environmental footprint of sofware, like Kuberne ...
+
+As a community, we always work to make each new release process better than before (in this release, we have [started to use Projects for tracking enhancements](https://github.com/orgs/kubernetes/projects/98/views/1), for example). If [v1.24 "Stargazer]"(/blog/2022/05/03/kubernetes-1-24-release-announcement/) _had us looking upwards, to what is possible when our community comes together_, and [v1 ...
```
Made the change manually on the updated version.

tengqm commented:
/label tide/merge-method-squash

fsmunoz commented:
> You might like to squash commits on this PR, @fsmunoz, at this point.

Done!

fsmunoz commented:
@sftim just updated the last piece missing (the webinar link), updated the stats and squashed.

sftim commented on content/en/blog/_posts/2022-12-08-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,355 @@
+
+### Availability
+
+Kubernetes v1.26 is available for download on
+[GitHub](https://github.com/kubernetes/kubernetes/releases/tag/v1.26.0). To get started with
+Kubernetes, check out these [interactive tutorials](/docs/tutorials/) or run local Kubernetes
+clusters using containers as "nodes", with [kind](https://kind.sigs.k8s.io/). You can also easily
+install v1.26 using [kubeadm](/docs/setup/independent/create-cluster-kubeadm/).
```
_Something for the release retro:_
these days, we could link to https://k8s.io/releases/download/

fsmunoz replied:
I can make that change if that's better, I used the same formula as previous releases but it might be time to update them.

sftim replied:
Your call @fsmunoz - I don't mind if _this_ release uses the old way.

sftim replied:
https://k8s.io/releases/download/ has details about how to validate that what you download has been signed.

leonardpahlke replied:
@kubernetes/release-managers does one might now more here?

sftim replied:
The [preview](https://deploy-preview-37810--kubernetes-io-main-staging.netlify.app/releases/download/) of https://k8s.io/releases/download/ looks broadly right although some details are stale AFAICT (eg: artifact signing is going to be beta)

fsmunoz replied:
Given the instructions on the artefact signing, which is an important theme in the release article itself, it might make sense to change it. Will push a commit with it, hadn't thought about it in that light.

sftim replied:
Cool, sounds good.

sftim commented:
/remove-label tide/merge-method-squash
/approve

kerthcet commented on content/en/blog/_posts/2022-12-08-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,354 @@
+[take taints/tolerations into consideration](/docs/concepts/scheduling-eviction/topology-spread-constraints/)
+when calculating Pod Topology Spread skew.
+
+#### `MatchLabelKeysInPodTopologySpread` graduates to beta
```
Sorry, this is still in alpha. https://github.com/kubernetes/enhancements/issues/3243, cc @alculquicondor for confirm.

alculquicondor replied:
that is correct, this is still alpha.

fsmunoz replied:
Isn't this this one https://github.com/kubernetes/enhancements/issues/3094 ? It is on the tracking board as graduating to beta.

kerthcet replied:
They're not the same, https://github.com/kubernetes/enhancements/issues/3094 is graduated to beta in this release, but not `MatchLabelKeysInPodTopologySpread`.

leonardpahlke replied:
[updated the stage](https://github.com/kubernetes/enhancements/issues/3094#issuecomment-1342837670).. thanks @kerthcet for your review!

leonardpahlke replied:
https://github.com/kubernetes/enhancements/issues/3243 is not part of the release

sftim replied:
```suggestion
#### Improved pod topology spread constraints
```
?

fsmunoz replied:
This was part of the [major themes discussion](https://github.com/kubernetes/sig-release/discussions/2047#discussioncomment-4162653) and was described as beta there, and as a major theme.  It was removed from milestone though, so I think this is a case where that wasn't captured. I would simply remove it since there is no movement in 1.26 related to it.

fsmunoz replied:
Done in last commit, will squash if there is no regression needed.

fsmunoz commented:
Updated file name, publication date and release date mentioned in the text following the delay to December 9.

sftim commented on content/en/blog/_posts/2022-12-09-kubernetes-1.26-blog.md:
```diff
@@ -0,0 +1,347 @@
+---
+layout: blog
+title: "Kubernetes v1.26: Electrifying"
+date: 2022-12-09
```
Looks right to me.

leonardpahlke commented:
/lgtm
/hold for @fsmunoz

leonardpahlke commented:
/lgtm

leonardpahlke commented:
@fsmunoz feel free to `/unhold` when ready

leonardpahlke commented:
/lgtm cancel
waiting for k/website freeze to be lifted https://github.com/kubernetes/website/issues/38317

fsmunoz commented:
I'm assuming that the doc links to alpha features will work when everything is merged, and if not I'll open up PRs to correct it.

/lgtm
/unhold

leonardpahlke commented:
/lgtm
